applications/StructuralMechanicsApplication/custom_elements/total_lagrangian.cpp

```diff
@@ -21,13 +21,71 @@
 #include "utilities/geometry_utilities.h"
 #include "structural_mechanics_application_variables.h"
 #include "custom_utilities/structural_mechanics_math_utilities.hpp"
+#include "utilities/geometrical_sensitivity_utility.h"
 
 namespace Kratos
 {
-TotalLagrangian::TotalLagrangian( IndexType NewId, GeometryType::Pointer pGeometry )
-        : BaseSolidElement( NewId, pGeometry )
+namespace
```

Comment:
Sometimes I want to add a helper function or class to a local cpp file, but it's either undesirable or too early to expose it for use by other parts of the code. In that case I put it in an unnamed name space which basically says it is only available in this file and the linker should not care if something with the same name exists in a different cpp file. I could have put e.g., CalculateGreenLagrangeStrainSensitivity in the class and avoided possible name conflicts in kratos namespace, but I didn't want to do that here because it would change the header (forcing more code to recompile) and because it doesn't depend in any way on the private data of the class. Eventually, such classes or functions could be move to a more public place, but initially when they appear after refactoring (such as for CalculateSensitivityMatrix in this case) and I'm not sure or ready to use them elsewhere, I put them here.

Reply:
Thanks for the detailed explanation, I was also wondering why you do this :+1:

Reply:
Hi @msandre can't we define a namespace "Detail" or "Internal" or something of the type for this? i think it is clearer than an unnamed namspace.

@pooyan-dadvand you are already using smthg of this type, right?

Reply:
We could do something like that to restrict the scope of such names. This is done in the hdf5 app, but it behaves differently from the unnamed namespace. I can also just remove the unnamed namespace since I don't think it will create a conflict here, but to me it's use is not a bad thing since it tells the reader something about the author's intent.